test/unit/math/rev/mat/fun/matrix_exp_action_test.cpp

```diff
@@ -0,0 +1,368 @@
+#include <stan/math/rev/mat.hpp>
+#include <gtest/gtest.h>
+// #include <stan/math/prim/scal/err/check_not_nan.hpp>
+#include <test/unit/math/rev/mat/fun/util.hpp>
+#include <test/unit/math/rev/mat/util.hpp>
+#include <stan/math/rev/mat/fun/matrix_exp_action_handler.hpp>
+#include <stan/math/rev/mat/fun/matrix_exp_action.hpp>
+#include <stan/math/rev/mat/fun/to_var.hpp>
+#include <vector>
+
+TEST(MathMatrix, matrix_exp_action_diag) {
+  using MatrixType = Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>;
+  using VectorType = Eigen::Matrix<double, Eigen::Dynamic, 1>;
+  stan::math::matrix_exp_action_handler handler;
+
+  {
+    double t = 0.5;
+    MatrixType m1(2, 2);
+    VectorType b(2);
+    m1 << 2, 0, 0, 2;
+    b << 1, 1;
+    auto res = handler.action(m1, b, t);
+    EXPECT_NEAR(res(0), M_E, 1.e-8);
```

Comment:
What is the `M_E` variable and where is it defined?

Reply:
It's built-in
https://msdn.microsoft.com/en-us/library/4hwaceh6.aspx